packages/js/k6/src/http.ts

```diff
@@ -0,0 +1,95 @@
+/**
+ * Wrap k6/http functions to sign requests to allow firewall bypass.
+ */
+
+import {
+  asyncRequest,
+  type RefinedResponse,
+  type RequestBody,
+  type RefinedParams,
+  type ResponseType,
+} from "k6/http"
+import { crypto, type CryptoKey } from "k6/experimental/webcrypto"
+
+// Polyfills URL
+import "core-js/web/url"
+
+import { b64encode } from "k6/encoding"
+
+let key: CryptoKey
+const signingSecret = __ENV.signing_secret
+
+async function getSigningKey(): Promise<CryptoKey | null> {
+  if (!signingSecret) {
+    return null
+  }
+
+  if (key) {
+    return key
+  }
+
+  const encodedSecret = stringToArrayBuffer(signingSecret)
+  key = await crypto.subtle.importKey(
+    "raw",
+    encodedSecret,
+    { name: "HMAC", hash: "SHA-256" },
+    false,
+    ["sign", "verify"]
```

Comment:
We don't need to verify with this key, so this can just be `sign`

```suggestion
    ["sign"]
```